addon/services/route-layers.js

```diff
@@ -2,7 +2,9 @@ import Ember from 'ember';
 
 export default Ember.Service.extend({
 
-  stack: [],
+  stack: Ember.computed(function() {
+    return Ember.A([]);
+  }),
```

Comment:
This was the potential bug exposed by testing this addon. Before, we were defining the stack array on the prototype, which can throw `calling set on destroyed object` errors. By making this a simple computed property, it resolves the issue.